paimon-core/src/main/java/org/apache/paimon/io/KeyValueFileWriterFactory.java

```diff
@@ -90,11 +97,21 @@ public DataFilePathFactory pathFactory() {
     }
 
     public RollingFileWriter<KeyValue, DataFileMeta> createRollingMergeTreeFileWriter(int level) {
+        updateFormatByLevel(level);
         return new RollingFileWriter<>(
                 () -> createDataFileWriter(pathFactory.newPath(), level, getCompression(level)),
                 suggestedFileSize);
     }
 
+    private void updateFormatByLevel(int level) {
```

Comment:
This style is not good because it dynamically decontaminates class member variables, which can lead to all sorts of problems.

In general, we recommend a functional style, which means that class members should ideally be immutable, with at most some cache stuff.

Reply:
I updated it.